src/lib/coda_base/pending_coinbase.ml

```diff
@@ -440,9 +692,9 @@ struct
         (Merkle_tree.get_req ~depth (Hash.var_to_hash_packed t) addr)
         reraise_merkle_requests
 
-    let%snarkydef add_coinbase t (pk, amount) =
-      let%bind addr =
-        request_witness Address.typ
+    let%snarkydef add_coinbase t (pk, amount, state_body_hash) =
+      let%bind addr, previous_state_hash, is_new_stack =
```

Comment:
We shouldn't have this unconstrained `is_new_stack` value -- we ought to be able to compute whether the stack is new.